### PR TITLE
Fix event repetition with errors on a branched pipeline

### PR DESCRIPTION
**Pull request by mfelsche**
# Pull request

## Description

This fixes an issue with a branched pipeline of which 1 branch (not the last) errors.
In that case the remaining branch events have not been removed and will be rexecuted with the next event, which leads to surprising behaviour, e.g. errors for the old event showing up after the next valid event has been handled.

## Checklist

* [x] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested
* [x] Use of unsafe code is reasoned about in a comment
* [x] Update CHANGELOG.md appropriately, recording any changes, bug fixes, or other observable changes in behavior
* [x] The performance impact of the change is measured (see below)

## Performance

Only the performance of the error case is affected, not _normal_ event processing.

### CODE REVIEW

File: tremor-pipeline/src/executable_graph.rs [hunk 1]
```diff
@@ -458,7 +458,16 @@ impl ExecutableGraph {
...
     #[inline]
     fn run(&mut self, returns: &mut Returns) -> Result<()> {
-        while stry!(self.next(returns)) {}
+        while match self.next(returns) {
+            Ok(res) => res,
+            Err(e) => {
+                // if we error handling an event, we need to clear the stack
+                // In the case of branching where the stack has > 1 element
+                // we would keep the old errored event around for the numbers of branches that havent been executed
+                self.stack.clear();
```
Comment (darach) [inline tremor-pipeline/src/executable_graph.rs:467]: ❤️  when a sleeping latent bug is tracked and whacked 🎉 - great catch @mfelsche

**Review by darach — DISMISSED** [review]
LGTM 👍

**Review by Licenser — APPROVED** [review]
Awesome, great catch Matthias! :rocket:

**Comment by mfelsche** [comment]
You did the catch, i did the typing :)

**Comment by Licenser** [comment]
No you!

**Pull request merged**